components/raftstore-v2/src/operation/life.rs

```diff
@@ -260,7 +306,223 @@ impl Store {
     }
 }
 
+/// Tell leader that target peer is destroyed.
+fn report_peer_destroyed(tombstone_msg: &mut RaftMessage) -> Option<RaftMessage> {
+    let to_region_id = if tombstone_msg.has_extra_msg() {
+        assert_eq!(
+            tombstone_msg.get_extra_msg().get_type(),
+            ExtraMessageType::MsgGcPeerRequest
+        );
+        tombstone_msg
+            .get_extra_msg()
+            .get_check_gc_peer()
+            .get_from_region_id()
+    } else {
+        tombstone_msg.get_region_id()
+    };
+    if to_region_id == 0 || tombstone_msg.get_from_peer().get_id() == 0 {
+        return None;
+    }
+    let mut msg = RaftMessage::default();
+    msg.set_region_id(to_region_id);
+    msg.set_from_peer(tombstone_msg.take_to_peer());
+    msg.set_to_peer(tombstone_msg.take_from_peer());
+    msg.mut_extra_msg()
+        .set_type(ExtraMessageType::MsgGcPeerResponse);
+    Some(msg)
+}
+
+fn tell_source_peer_to_destroy<EK, ER, T>(ctx: &mut StoreContext<EK, ER, T>, msg: &RaftMessage)
```

Comment:
merge_source_peer, source/target can be confused with msg sender/recipient.

Reply:
source/target is referring merge within this project unless stated explicitly.

Reply:
You used target to represent message receiver at L309.

Reply:
+1. There's no hurt to make it more explicit as this file is not about merge.  Or, at least have a comment to clarify source_peer is merge source peer.

Reply:
Also regarding the naming, tell is a bit too oral and maybe request is better.

Reply:
I have already update the `target` usage at L309. `source peer` and `target peer` are very frequent within in the project. It's more ambiguous to make it represent different meanings.

Reply:
`tell` is actually a technical term in actor. But I will change it to `forward` instead.

Reply:
Even in existing code, it is rare to find usage of "source" directly without hinting the context of merge. Here this function alone is confusing to grasp because neither the implementation nor the function name implies such context. You have to trace it back to caller to get the context, which is not ideal. I think here the least you could do is adding some comments.

Reply:
> it is rare to find usage of "source" directly without hinting the context of merge.

I think this explains "source/target is referring merge within this project unless stated explicitly" perfectly. A comment is also OK.